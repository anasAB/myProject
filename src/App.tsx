import './App.css';
import Chop from './chop';
import { IDataset } from './Typing/Dataset';


const dataset: IDataset = {
  bread: "baguette",
  cheese: "brie cheese",
};



const myTemplate: string = `I like to eat ${dataset.bread} and ${dataset.cheese}`;


function App() {

  let dataSetError;
  if (typeof dataset.bread !== 'string' || typeof dataset.cheese !== 'string') {
    return dataSetError = <p>Please the Enterd data should be String</p>
  }


  return (
    <div className="App">
      <header className="App-header">
        <p>Yara</p>
        <p>{dataSetError}</p>
        <Chop data={dataset} template={myTemplate} />
      </header>
    </div>
  );
}

export default App;
