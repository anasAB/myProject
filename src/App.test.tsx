import { shallow } from 'enzyme';
import App from './App';
import Chop from './chop';
import renderer from 'react-test-renderer';


const MockDataset = {
  bread: "baguette",
  cheese: "brie cheese",
};

const myTemplate: string = `I like to eat ${MockDataset.bread} and ${MockDataset.cheese}`;

describe('rendering App', () => {

  it('should render correctly', () => {
    const AppWrapper = shallow(<App />);
    console.log('AppWrapper', AppWrapper);
    expect(AppWrapper).toMatchSnapshot();
  })

  it('should render chop', () => {
    const tree = renderer.create(<Chop data={MockDataset} template={myTemplate} />).toJSON()
    expect(tree).toMatchSnapshot();
    // expect(tree).tobe
    console.log('tree', tree);
  })


  //**! with correct props */
  it('should render chop with chop element', () => {
    const wrapper = shallow(<Chop data={MockDataset} template={myTemplate} />);
    expect(wrapper.contains(<p>{myTemplate}</p>)).toBeTruthy();

  })

  //**! missing one of the props */
  it('should render the error Message', () => {
    const MockEmptyDataset = null
    const wrapper = shallow(<Chop data={MockEmptyDataset} template={myTemplate} />);
    expect(wrapper.contains(<p>template and dataset are mandatory parameters</p>)).toBeTruthy();
  })


});

