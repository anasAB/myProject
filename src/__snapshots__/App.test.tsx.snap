// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering App should render chop 1`] = `
<div
  style={
    Object {
      "color": "green",
    }
  }
>
  <p>
    I like to eat baguette and brie cheese
  </p>
</div>
`;

exports[`rendering App should render correctly 1`] = `ShallowWrapper {}`;
